core/src/main/scala/kafka/server/ReplicationQuotaManager.scala

```diff
@@ -114,14 +115,17 @@ class ReplicationQuotaManager(val config: ReplicationQuotaManagerConfig,
   }
 
   /**
+    * Is the broker throttled by this ReplicationQuotaManager or
     * Is the passed partition throttled by this ReplicationQuotaManager
     *
     * @param topicPartition the partition to check
     * @return
     */
   override def isThrottled(topicPartition: TopicPartition): Boolean = {
     val partitions = throttledPartitions.get(topicPartition.topic)
-    if (partitions != null)
+    if (isBrokerThrottled)
```

Comment:
I think it'd be good to go through a formal discussion (via KIP and mailing list) on how this new config would apply. There may be benefits in having `throttled=true, replicas=1:0,2:0,3:0` apply to the certain replicas only. I personally don't see them though and I think this solution is the better one. Worth discussing the alternatives anyway